descope/types.go

```diff
@@ -1034,17 +1034,32 @@ type ThirdPartyApplicationConsent struct {
 	GrantedBy string `json:"grantedBy"`
 	// The time the consent was granted in milliseconds since epoch
 	CreatedTime int32 `json:"createdTime"`
+	// The tenant id the consent was granted for
+	TenantID string `json:"tenantId"`
 }
 
 // Options for deleting third party application consents. At least one of ConsentIDs, AppID or UserIDs must be provided.
 //
+// ConsentIDs - allows to delete any consents by their given id.
 // AppID - allows to delete all consents by a given third party application id.
 // UserID - allows to delete all consents of a given user by user id.
-// ConsentIDs - allows to delete any consents by their given id.
+// TenantID - allows to delete all consents of a given tenant by tenant id.
```

Comment:
Shouldn’t we just allow tenant ID only on the search by tenant function?

Reply:
Nop, as this function used for delete all scenarios of tenant-level consents like "delete all my app AAA tenant-level consent" so in such case we would like to give the appId as the key..